src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1118,51 +1117,108 @@ def add_units_table(
-                f"Description for property {property_name} not found in property_descriptions. "
-                "Setting description to 'no description'"
-            )
-    aggregated_unit_properties = defaultdict()
-    for property_name in write_properties:
-        unit_col_args = dict(
-            name=property_name, description=property_descriptions.get(property_name, "No description.")
-        )
-        if property_name in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
-            unit_col_args.update(table=nwbfile.electrodes)
+    data_to_add = defaultdict(dict)
+    sorting_properties = checked_sorting.get_property_keys()
+    excluded_properties = list(skip_properties) + ["contact_vector"]
+    properties_to_extract = [property for property in sorting_properties if property not in excluded_properties]
 
-        units_table.add_column(**unit_col_args)
-        aggregated_unit_properties[property_name] = checked_sorting.get_property(key=property_name)
+    # Extract properties
+    for property in properties_to_extract:
+        data = checked_sorting.get_property(property)
+        index = isinstance(data[0], (list, np.ndarray, tuple))
+        description = property_descriptions.get(property, "No description.")
+        data_to_add[property].update(description=description, data=data, index=index)
+        if property in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
+            data_to_add[property].update(table=nwbfile.electrodes)
 
     # Unit name logic
     units_ids = checked_sorting.get_unit_ids()
-    if "unit_name" in aggregated_unit_properties:
-        unit_name_array = aggregated_unit_properties["unit_name"]
+    if "unit_name" in data_to_add:
+        unit_name_array = data_to_add["unit_name"]["data"]
     else:
         unit_name_array = units_ids.astype("str", copy=False)
-        unit_col_args = dict(
-            name="unit_name",
-            description="unique reference for each unit",
-        )
-        units_table.add_column(**unit_col_args)
-        aggregated_unit_properties["unit_name"] = unit_name_array
+        data_to_add["unit_name"].update(description="unique reference for each unit", data=unit_name_array, index=False)
 
-    for i, unit_id in enumerate(units_ids):
-        spkt = checked_sorting.get_unit_spike_train(unit_id=unit_id, return_times=True)
-        kwargs = {key: val[i] for key, val in aggregated_unit_properties.items()}
+    # If the channel ids are integer keep the old behavior of asigning table's id equal to unit_ids
+    if np.issubdtype(units_ids.dtype, np.integer):
+        data_to_add["id"].update(data=units_ids.astype("int"), index=False)
 
-        # If the units ids are integer keep the old behavior of asigning table's id equal to unit_ids
-        if str(unit_id).isdigit():
-            id = int(unit_id)
-        else:
-            id = i
+    units_table_previous_properties = set(units_table.colnames) - {"spike_times"}
```

Comment:
Is "spike_times" all you need or to you also need "spike_times_index"?

Reply:
Also, using literals wherever possible helps readability. *E.g.*, `{"spike_times"}` -> `set(["spike_times"])`. The extra computational time from casting the list ought to be negligible. Otherwise it can be easy to confuse on quick glance with a `dict`.

Reply:
I personally find the notation of the left more readable but I have been using sets for two long. What you say sounds true: someone seeing the word set as in the literal is better for avoiding the confusion. I am curios on why python would not allow a directly set construction like `set(element_1, element_2, element_3)` probably hard to parse compared to the normal construction `set(list)`.